### ISSUE TITLE
riverpod_generator produces InvalidType when using graphql_codegen generated types as parameter

### ISSUE THREAD

**Issue by vipulog** [rrousselGit/river_pod#3196]
**Describe the bug**
graphql_codegen is a code-generation tool which generates dart serializers from graphql. When using any type generated by grapql_codegen in a provider, the analyzer shows Undefined class 'InvalidType'. error in file generated by riverpod_generator. 

**To Reproduce**
1. Add graphql _codegen:
```
flutter pub add --dev graphql_codegen build_runner
```

2. Add schema:
```
enum MediaFormat {
  TV
  TV_SHORT
  MOVIE
}
```

3. Add Provider:
```
@riverpod
void demo(DemoRef ref, Enum$MediaFormat format) {}
```

**Comment by rrousselGit** [comment]
Yes, this is a known issue. It has been fixed in the 3.0 dev release

**Comment by mrice-wisecode** [comment]
I've been unable to migrate from Riverpod 2.0 -> 3.0 and believe the reason is my providers will not generate because of graphql_codegen classes as the return type. @rrousselGit any chance this did not actually get fixed or is a known issue? I couldn't find any other issues for it
